internal/zinc-compile-core/src/main/scala/sbt/internal/inc/AnalyzingCompiler.scala

```diff
@@ -239,7 +239,7 @@ final class AnalyzingCompiler(
       log: Logger
   )(argTypes: Class[_]*)(args: AnyRef*): AnyRef = {
     val interfaceClass = getInterfaceClass(interfaceClassName, log)
-    val interface = interfaceClass.newInstance.asInstanceOf[AnyRef]
+    val interface = interfaceClass.getDeclaredConstructor().newInstance().asInstanceOf[AnyRef]
```

Comment:
Does this have the same semantics in JDK 8?

Reply:
Yes. `getDeclaredConstructor().newInstance()` is same as calling `new` with the nullary constructor, which is what `Class#newInstance()` did but it's now deprecated.  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Class.html#newInstance()